### PR TITLE
Rename settings to host

### PR DESCRIPTION
**Pull request by ahuang11**
Converts settings.DockerSettings to host.DockerHost.

I removed credstore_env from DockerSettings because [docs were inaccurate](https://docker-py.readthedocs.io/en/stable/client.html#docker.client.from_env) about it being available for from_env.

```
File ~/mambaforge/lib/python3.10/site-packages/docker/client.py:101, in DockerClient.from_env(cls, **kwargs)
    100     use_ssh_client=use_ssh_client,
--> 101     **kwargs_from_env(**kwargs)
    102 )

TypeError: kwargs_from_env() got an unexpected keyword argument 'credstore_env'
```

I also stripped the settings kwargs to remove keys that had values of None to make it work.

Closes #

### Checklist

- [x] This pull request references any related issue by including "Closes #<ISSUE_NUMBER>"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-docker/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] Summarized PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-docker/blob/main/CHANGELOG.md)

### CODE REVIEW

File: prefect_docker/host.py [hunk 1]
```diff
@@ -42,14 +40,6 @@ class DockerSettings(Block):
...
         description="The maximum number of connections to save in the pool.",
     )
-    credstore_env: Dict[str, Any] = Field(
-        default_factory=dict,
-        title="Credential Store Environment Variables",
-        description=(
-            "Override environment variables when calling "
-            "the credential store process."
-        ),
-    )
```
Comment (desertaxle) [inline prefect_docker/host.py:52]: Is `credstore_env` a valid kwarg when instantiating a Docker client with a base URL?

File: CHANGELOG.md [hunk 2]
```diff
@@ -25,4 +25,8 @@ Released on ????? ?th, 20??.
 
 ### Added
 
-- `DockerSettings` task - [#3](https://github.com/PrefectHQ/prefect-docker/pull/3)
+- `DockerSettings` block - [#3](https://github.com/PrefectHQ/prefect-docker/pull/3)
+
+### Changed
+
+- `DockerSettings` renamed to `DockerHost` - [#13](https://github.com/PrefectHQ/prefect-docker/pull/13)
```
Comment (desertaxle) [inline CHANGELOG.md:32]: Just a comment, doesn't need to change in this PR: when we release this collection, I think we should consolidate these two entries into one entry under just Added.

File: prefect_docker/host.py (hunk 1 above)
Comment (ahuang11) [reply prefect_docker/host.py:52]: It is.

File: CHANGELOG.md (hunk 2 above)
Comment (ahuang11) [reply CHANGELOG.md:32]: Which PR number should I reference?

Comment (desertaxle) [reply CHANGELOG.md:32]: I think we can reference both

**Review by desertaxle — APPROVED** [review]

**Pull request merged**